### DISCUSSION TITLE
General

### DISCUSSION THREAD

**Discussion by chris-ritsen** [chris-ritsen/network-audio-controller#4] [General]
General discussion

**Comment by foxmox8** [comment]
Hello Chris,

Thanks a lot for your python package.
I'm struggling to use netaudio within a script... like, I would like to use a function to call the "Channel List" option in netaudio (for example) 

Is it possible to have a little bit of help on this subject please ? 

Thanks a lot

F8

**Comment by ccaudle** [comment]
I just recently discovered this project, and am trying to get started evaluating.  I cloned the git repository locally, then followed the instructions for using poetry to run without installing, and get this exception after poetry run netaudio:
  File "/home/chris/Src/NetworkAudioController/network-audio-controller/netaudio/console/commands/subscription/_list.py", line 13, in <module>
    from netaudio.dante.cache import DanteCache
ModuleNotFoundError: No module named 'netaudio.dante.cache'

I tried to find any DanteCache definition, and only found where it was attempted to use, no class or function definitions:
$ find ./ -name "*.py" | xargs grep DanteCache
./netaudio/console/commands/subscription/_list.py:from netaudio.dante.cache import DanteCache
./netaudio/console/commands/subscription/_list.py:            dante_cache = DanteCache()

So the only reference to DanteCache in the repository seems to be in _list.py.  Did I miss cloning some sub-repository needed?
I just used git clone with the URL provided by the "Code" drop  down on the GitHub project page.

**Reply by chris-ritsen** [reply]
You didn't miss anything, it's just half-baked, partially implemented ideas on how to architect this project. My intent is to have some kind of cache so it doesn't have to do anything with mDNS service lookups each time the program runs to make it faster for scripting.

**Comment by fadersolo** [comment]
Hello, I’m new to Dante, but have been running two studios based on Soundgrid for several years, but am switching my home studio over to Dante/AES67.

So far, coming from Soundgrid, Dante seems both amazing and sometimes confusing. The most frustrating thing I’m encountering is what happens when I load projects in different sample rates. I’m a grammy nominated recording engineer, make most of my money mixing in stereo, but also have a part time job composing music video game studio (and for the next two years it’ll all be in 5.1). Most of the projects I’m in charge of are at 48k, but so many of my clients send me projects to mix at other sample rates. I’ve gotten used to not thinking about it, but now when I open a session at 44.1k...there’s no graceful way to simply switch everything over to the new sample rate. Until I found this project! Now it’s as simple as running `netaudio config --set-sample-rate 44100`. Well, almost as simple, I’m having issues which I’ve shared [over here](https://github.com/chris-ritsen/network-audio-controller/issues/26). But I gotta say, this is a really exciting project and I’m super excited to be making progress automating some of the Dante config without having to deal with that awkward Dante Controller.
